src/analysis/vector/geometry_checker/qgsgeometrymissingvertexcheck.cpp

```diff
@@ -0,0 +1,119 @@
+/***************************************************************************
+    qgsgeometrymissingvertexcheck.cpp
+    ---------------------
+    begin                : September 2018
+    copyright            : (C) 2018 Matthias Kuhn
+    email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsgeometrymissingvertexcheck.h"
+
+#include "qgsfeaturepool.h"
+#include "qgsgeometrycollection.h"
+#include "qgscurvepolygon.h"
+#include "qgscurve.h"
+#include "qgslinestring.h"
+#include "qgsgeometryengine.h"
+#include "qgsgeometryutils.h"
+
+void QgsGeometryMissingVertexCheck::collectErrors( QList<QgsGeometryCheckError *> &errors, QStringList &messages, QAtomicInt *progressCounter, const QMap<QString, QgsFeatureIds> &ids ) const
```

Comment:
Unrelated comment: I think a good API addition would be to have a QgsFeedback argument here -- this would allow progress reports and cancelation.

Reply:
Already implemented in another branch.